Framework/Sources/FloatingPanel.swift

```diff
@@ -477,24 +514,95 @@ class FloatingPanel: NSObject, UIGestureRecognizerDelegate {
+                self.startNewAnimator(for: distance, with: velocity, movingTo: newTargetState)
+            }
+
+            // Since the drag is already in progress we need 'save' the already travelled distance on the screen in
+            // order to calculate the correct translation when calcualting the fraction (-> progress) of the animator.
+            // Otherwise the panel 'jumps' from its current position to the finger's position on the screen.
+            self.alreadyMovedTranslationOfPan = translation
+        }
+
+        self.preserveContentVCLayoutIfNeeded()
+        // Determine whether or not the panel has actually moved.
+        if animator.fractionComplete != previousFractionComplete {
+            viewcontroller!.delegate?.floatingPanelDidMove(viewcontroller!)
+        }
+    }
+
+    private func startNewAnimator(for distance: CGFloat, with velocity: CGPoint, movingTo targetState: FloatingPanelPosition) {
+        let velocityVector = (distance != CGFloat(0)) ? CGVector(dx: 0, dy: min(abs(velocity.y)/distance, 30.0)) : .zero
+
+        let animator = behavior.interactionAnimator(self.viewcontroller!, between: self.state, and: targetState, with: velocityVector)
+        animator.addAnimations { [weak self] in
+            guard let `self` = self else { return }
+            // Set layout (-> therefor the top constraint of the FloatingPanel) to the next upcoming state in order to
+            // be able to move the panel during a drag
+            self.updateLayout(to: targetState)
+        }
+        animator.addCompletion { [weak self] pos in
+            guard let `self` = self else { return }
+            if pos == .end, let animator = self.animator, !animator.isReversed {
+                self.state = self.nextState
+            }
+            // Set layout again to ensure that the panel's top constraint corresponds with the desired state.
+            self.updateLayout(to: self.state)
+            self.animationDidComplete(at: self.state)
+        }
+
+        if self.state == targetState {
+            return
+        }
+
+        self.animator = animator
+        self.nextState = targetState
+
+        // In order to make animators drag-dynamic we need to start and immediatly pause the animator aftwards.
+        animator.startAnimation()
+        animator.pauseAnimation()
+
+        // Calculate the total distance the floating panel has to move between its current state and its next state.
+        // Depending on the direction of the drag we need turn the calcualtion around in order to still get a positive
+        // value for totalYTranslation.
+        if velocity.y < CGFloat(0) {
+            self.totalYTranslation = abs(self.layoutAdapter.positionY(for: self.state) - self.layoutAdapter.positionY(for: nextState))
+        } else if velocity.y > CGFloat(0) {
+            self.totalYTranslation = abs(self.layoutAdapter.positionY(for: nextState) - self.layoutAdapter.positionY(for: self.state))
         }
```

Comment:
Is it possible that `velocity.y` is zero when this method is called?

Reply:
When I tested it, it was sometimes zero and sometimes it wasn't.